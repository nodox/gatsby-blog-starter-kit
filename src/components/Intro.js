import React from 'react';
import PropTypes from 'prop-types';

export default class Intro extends React.Component {
  constructor(props) {
    super(props)
  }

  getDescription(currentPath, pages) {
    for (var key in pages) {
      if (currentPath === pages[key]['path']) {
        return pages[key].introText;
      }
    }
    return '';
  }

  render() {
    const {
      currentPath,
      config,
    } = this.props;

    const {
      name,
      pages,
    } = config;

    const description = this.getDescription(currentPath, pages);

    return (
      <div id="intro" className="">
        <div className="img-container">
          <div className="img-size-sm img-intro img-circle"></div>
        </div>
        <h1>{name}</h1>
        <p>{description}</p>
        <ul className="actions">
          <li>
            <a href="#header" className="button icon solo fa-arrow-down scrolly">Continue</a>
          </li>
        </ul>
      </div>
    );
  }
}
