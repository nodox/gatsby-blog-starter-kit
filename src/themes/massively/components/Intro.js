import React from 'react';
import PropTypes from 'prop-types';

export default class Intro extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div id="intro" className="">
        <div className="img-container">
          <div className="img-size-sm img-intro img-circle"></div>
        </div>
        <h1>{this.props.name}</h1>
        <p>{this.props.description}</p>
        <ul className="actions">
          <li>
            <a href="#header" className="button icon solo fa-arrow-down scrolly">Continue</a>
          </li>
        </ul>
      </div>
    );
  }
}
