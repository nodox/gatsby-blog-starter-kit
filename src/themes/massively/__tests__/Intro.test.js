import React from 'react';
import renderer from 'react-test-renderer';
import Intro from '../components/Intro';

describe('<Intro />', () => {
  const data = 'Steven Natera';

  const component = renderer.create(
    <Intro description={data} />
  );

  const props = component.getInstance().props;

  it('should render correctly', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have a the right props', () => {
    expect(props).toHaveProperty('description');
  });

});
