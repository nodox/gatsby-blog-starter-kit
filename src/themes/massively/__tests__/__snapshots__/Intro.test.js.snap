// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> should render correctly 1`] = `
<div
  className=""
  id="intro"
>
  <div
    className="img-container"
  >
    <div
      className="img-size-sm img-intro img-circle"
    />
  </div>
  <h1>
    Steven Natera
  </h1>
  <p />
  <ul
    className="actions"
  >
    <li>
      <a
        className="button icon solo fa-arrow-down scrolly"
        href="#header"
      >
        Continue
      </a>
    </li>
  </ul>
</div>
`;
